src/function/built_in_vector_operations.cpp

```diff
@@ -73,6 +73,56 @@ std::vector<std::string> BuiltInVectorOperations::getFunctionNames() {
     return result;
 }
 
+uint32_t BuiltInVectorOperations::getCastCost(DataTypeID inputTypeID, DataTypeID targetTypeID) {
+    if (inputTypeID == targetTypeID) {
+        return 0;
+    } else {
+        if (targetTypeID == ANY) {
+            // Any inputTypeID can match to type ANY
+            return 0;
+        }
+        switch (inputTypeID) {
+        case common::ANY:
+            // ANY type can be any type
+            return 0;
+        case common::INT64:
+            return implicitCastInt64(targetTypeID);
+        case common::DOUBLE:
+            return implicitCastDouble(targetTypeID);
+        default:
+            return UINT32_MAX;
+        }
+    }
+}
+
+uint32_t BuiltInVectorOperations::getTargetTypeCost(common::DataTypeID targetTypeID) {
+    switch (targetTypeID) {
+    case common::DataTypeID::INT64:
+        return 103;
+    case common::DataTypeID::DOUBLE:
+        return 102;
+    default:
+        throw InternalException(
+            "Undefined cast cost to type " + common::Types::dataTypeToString(targetTypeID));
+    }
+}
+
+uint32_t BuiltInVectorOperations::implicitCastInt64(common::DataTypeID targetTypeID) {
+    switch (targetTypeID) {
+    case common::DOUBLE:
+        return getTargetTypeCost(targetTypeID);
```

Comment:
I would not introduce getTargetTypeCost in this PR. Just return 102/1 in this case because we only allow cast int64 -> double

Reply:
Directly returned 102 here and removed getTargetTypeCost.